tooling/profiler/src/errors.rs

```diff
@@ -0,0 +1,15 @@
+use fm::FileMap;
+use noirc_errors::{CustomDiagnostic, Span};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub(crate) enum CliError {
+    #[error("Failed to run profiler command")]
+    Generic,
+}
+
+pub(crate) fn report_error(message: String) -> Result<(), CliError> {
+    let error = CustomDiagnostic::simple_error(message.clone(), String::new(), Span::default());
+    noirc_errors::reporter::report(&FileMap::default(), &error, None, false);
```

Comment:
Does the `FileMap::default()` work on errors from dependencies in multi-crate projects? I would expect it to need to be initialized for accurate reporting, as is done [here](https://github.com/noir-lang/noir/blob/1d04f8ba0292e493e4e64cf8caf6df15c51b3346/tooling/nargo_cli/src/cli/execute_cmd.rs#L172C27-L172C61).

Reply:
The profiler uses binary artifacts, so a user is going to have to point to the correct artifact if they are working within a workspace. I did `FileMap::default()` as this allows us to report in cases where we do not have a stack trace and avoiding what is done [here](https://github.com/noir-lang/noir/blob/1d04f8ba0292e493e4e64cf8caf6df15c51b3346/tooling/nargo_cli/src/cli/execute_cmd.rs#L172C27-L172C61).

I'll add a comment.